x-pack/test/functional/services/transform/date_picker.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function TransformDatePickerProvider({ getService, getPageObjects }: FtrProviderContext) {
+  const find = getService('find');
+  const testSubjects = getService('testSubjects');
+  const pageObjects = getPageObjects(['timePicker']);
+
+  return {
+    async assertSuperDatePickerToggleQuickMenuButtonExists() {
+      await testSubjects.existOrFail('superDatePickerToggleQuickMenuButton');
+    },
+
+    async openSuperDatePicker() {
+      await testSubjects.click('superDatePickerToggleQuickMenuButton');
+      await testSubjects.existOrFail('superDatePickerQuickMenu');
+    },
+
+    async quickSelect(timeValue: number = 15, timeUnit: string = 'y') {
+      const quickMenuElement = await testSubjects.find('superDatePickerQuickMenu');
+
+      // No test subject, defaults to select `"Years"` to look back 15 years instead of 15 minutes.
+      await find.selectValue(`[aria-label*="Time value"]`, timeValue.toString());
+      await find.selectValue(`[aria-label*="Time unit"]`, timeUnit);
+
+      // Apply
+      const applyButton = await quickMenuElement.findByClassName('euiQuickSelect__applyButton');
+      const actualApplyButtonText = await applyButton.getVisibleText();
+      expect(actualApplyButtonText).to.be('Apply');
+
+      await applyButton.click();
```

Comment:
Would be good to add an assertion after button is clicked to check that the time range was indeed updated

Reply:
Updated in [30f7f7f](https://github.com/elastic/kibana/pull/149049/commits/30f7f7f0de364671ac63f0273cf4d4382d72db16).